### PR TITLE
feature/stats: add subject type

### PR DESCRIPTION
**Pull request by benoithebrard**
- a notification may contain a subject Id and subject type

### CODE REVIEW

File: notifications/src/main/java/dk/shape/games/notifications/entities/Subscription.kt [hunk 1]
```diff
@@ -8,6 +8,8 @@ import kotlinx.serialization.Serializable
  */
 @Serializable
 data class Subscription(
-    @SerialName("event_id") val eventId: String,
-    @SerialName("types") val types: Set<String>
+    @SerialName("event_id") val eventId: String? = null,
+    @SerialName("subject_id") val subjectId: String? = null,
+    @SerialName("subject_type") val subjectType: SubjectType? = null,
+    @SerialName("types") val types: Set<String> = setOf()
 )
```
Comment (Shaper-Eudy-Contreras) [inline notifications/src/main/java/dk/shape/games/notifications/entities/Subscription.kt:15]: Can we not have a polymorphic subscription instead. 

```kotlin
data class Subscription.Stats(
    @SerialName("subject_id") val subjectId: String,
    @SerialName("subject_type") val subjectType: SubjectType,
    @SerialName("types")  override val types: Set<String> = setOf()
) : Subscription(types = types)

data class Subscription.Events(
    @SerialName("event_id") val eventId: String,
    @SerialName("types")  override val types: Set<String> = setOf()
) : Subscription(types = types)
```
We can also deal with deserialisation separately

File: notifications/src/main/java/dk/shape/games/notifications/entities/SubjectType.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.google.gson.annotations.SerializedName
+
+/**
+ * Entity representing all subject types that can be used for stats notifications.
+ */
+enum class SubjectType {
+    @SerializedName("events")
+    EVENTS,
+
```
Comment (Shaper-Eudy-Contreras) [inline notifications/src/main/java/dk/shape/games/notifications/entities/SubjectType.kt:11]: Why should event be a possible subject type?

Comment (benoithebrard) [reply notifications/src/main/java/dk/shape/games/notifications/entities/SubjectType.kt:11]: that's according to that spec: https://docs.shapegamesbwtest.com/sports/notifications.html#/paths/~1subscriptions/get It was added by the backend team to help them sort out notifications data source

**Comment by benoithebrard** [comment]
I get the error `dk.shape.games.notifications.entities.SubjectType does not contain element with name 'events'` Could you have a look at it @Shaper-Eudy-Contreras ? Otherwise I'll have to abandon this PR

**Comment by Shaper-Eudy-Contreras** [comment]
> I get the error `dk.shape.games.notifications.entities.SubjectType does not contain element with name 'events'` Could you have a look at it @Shaper-Eudy-Contreras ? Otherwise I'll have to abandon this PR

I'll look at it :raised_hands:

**Comment by benoithebrard** [comment]
Now it's working fine @Shaper-Eudy-Contreras so this PR is ready for review again

**Pull request merged**